resources/centos.rb

```diff
@@ -0,0 +1,107 @@
+resource_name :osl_repos_centos
+provides :osl_repos_centos
+
+default_action :add
+
+# These properties indicate wether or not a repo should be managed
+# If unmanaged the repo file will not be created or changed
+property :base, [true, false], default: true
+property :extras, [true, false], default: true
+property :updates, [true, false], default: true
+
+# These repositories are only availible on Centos 8
+# The associated properties will be ignored on Centos 7
+property :appstream, [true, false], default: true
+property :powertools, [true, false], default: true
+
+# These properties indicate wether or not a repo should be enabled
+# Note: If unmanaged is set the repo's enabled/disabled state will not be changed
+property :base_enabled, [true, false], default: true
+property :extras_enabled, [true, false], default: true
+property :updates_enabled, [true, false], default: true
+
+# These repositories are only availible on Centos 8
+# The associated properties will be ignored on Centos 7
+property :appstream_enabled, [true, false], default: true
+property :powertools_enabled, [true, false], default: true
+
+# This is the default and only action, It will add all availible repos, unless specified in properties above
+action :add do
+  # Set 'installonly_limit' and 'installonlypkgs' in the main yum configuration file.
+  node.default['yum']['main']['installonly_limit'] = '2'
+  node.default['yum']['main']['installonlypkgs'] = 'kernel kernel-osuosl'
+
+  # Initialize all repo mirrorlists to nil
+  # Note: any changes made here and throught the recipe will not apply if the corresponding repository is unmanaged
+  node.default['yum']['appstream']['mirrorlist'] = nil
+  node.default['yum']['base']['mirrorlist'] = nil
+  node.default['yum']['extras']['mirrorlist'] = nil
+  node.default['yum']['powertools']['mirrorlist'] = nil
+  node.default['yum']['updates']['mirrorlist'] = nil
+
+  # As mentioned above C7 and C8 have different availible repos and options
+  case node['platform_version'].to_i
+
+  ####### Begin Centos 7 case #######
+  when 7
+
+    # Update the base url for our Centos 7 repos (base, updates, extras, and epel)
+    node.default['yum']['base']['baseurl'] = "#{centos_url}/$releasever/os/#{base_arch}/"
+    node.default['yum']['updates']['baseurl'] = "#{centos_url}/$releasever/updates/#{base_arch}/"
+    node.default['yum']['extras']['baseurl'] = "#{centos_url}/$releasever/extras/#{base_arch}/"
+
+    # Set the gpg key for each repo (excluding epel)
+    # This case ensures that the AltArch gpg key is included in any non x86_64 architecture
+    %w(base updates extras).each do |r|
+      node.default['yum'][r]['gpgkey'] =
+        if node['kernel']['machine'] == 'x86_64'
+          'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7'
+        else
+          'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7 ' \
+          'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-AltArch-7-$basearch'
```

Comment:
This should be a better way than what we're doing now.
```suggestion
          'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7 file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-AltArch-7-$basearch'
```